sycl/include/sycl/handler.hpp

```diff
@@ -3394,17 +3420,35 @@ class __SYCL_EXPORT handler {
   ///        expr m_Storage member
   /// \param Size the size of data getting read back / to.
   /// \param Block if read operation is blocking, default to false.
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+  void ext_intel_read_host_pipe(const std::string &Name, void *Ptr, size_t Size,
+                                bool Block = false) {
+    ext_intel_read_host_pipe(detail::string_view(Name), Ptr, Size, Block);
+  }
+  void ext_intel_read_host_pipe(detail::string_view Name, void *Ptr,
+                                size_t Size, bool Block = false);
```

Comment:
I'm surprised this has the same visibility (public/private) as the one above. Can you explain why is this a correct thing to do?

Reply:
We cannot pass a `std::string` from the header to the libsycl side due to the ABI incompatibility.

Reply:
So what? `detail::string` isn't part of any *public* interface, is it?

Reply:
Oh. sorry, I misunderstood your comment.
I thought you were asking why we were changing from `std::string` paramter to `std::string_view` parameter.
Yes, I can put it in the private visibility.
Thanks for clarification.

Reply:
> line 3428 is already in the private section.

That's surprising to me because of the name, but if so, then nothing to do here.